apps/btc_family/btc_fam_main.c

```diff
@@ -0,0 +1,137 @@
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "btc_fam_main.h"
+
+#include "btc_fam_api.h"
+#include "btc_fam_priv.h"
+#include "main_menu.h"
+#include "status_api.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+const btc_fam_config_t *g_app = NULL;
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+void btc_fam_main(usb_event_t usb_evt, const btc_fam_config_t *app) {
+  btc_query_t query = BTC_QUERY_INIT_DEFAULT;
+  g_app = app;
+
+  if (false == decode_btc_query(usb_evt.p_msg, usb_evt.msg_size, &query)) {
+    return;
+  }
+
+  /* Set status to CORE_DEVICE_IDLE_STATE_USB to indicate host that we are now
+   * servicing a USB initiated command */
+  core_status_set_idle_state(CORE_DEVICE_IDLE_STATE_USB);
+
+  LOG_SWV("%s (%d) - Query:%d\n", __func__, __LINE__, query.which_request);
+  switch ((uint8_t)query.which_request) {
+    case BTC_QUERY_GET_PUBLIC_KEY_TAG: {
+      btc_fam_pub_key(&query);
+      break;
+    }
+    case BTC_QUERY_GET_XPUBS_TAG: {
+      btc_fam_xpub(&query);
+      break;
+    }
+    default: {
+      /* In case we ever encounter invalid query, the USB event should be
+       * cleared manually */
+      usb_clear_event();
```

Comment:
```suggestion
      /* In case we ever encounter invalid query, convey to the host app */
      btc_send_error(ERROR_COMMON_ERROR_CORRUPT_DATA_TAG,
                     ERROR_DATA_FLOW_INVALID_QUERY);
```